### ISSUE TITLE
Delay between release and container publish

### ISSUE THREAD

**Issue by mmattel** [zwave-js/zwave-js-ui#3430]
Q: when there is a new release like in the screenshot below, what is the expected update on hub.docker to make the release availabe?

**Repo**
![image](https://github.com/zwave-js/zwave-js-ui/assets/3321281/6de6c2d4-9d4c-45ce-a491-08ff025112b1)
**hub.docker**
![image](https://github.com/zwave-js/zwave-js-ui/assets/3321281/2de40767-8889-4afc-83f5-d5cef4b03b77)

**Comment by myoung34** [comment]
This also bit me [here](https://github.com/myoung34/homelab/commit/1a7cb51e0e92068509ebfa0e75a775aa1e78793a)
It picks up releases but this project seems to consistently not publish

* https://github.com/zwave-js/zwave-js-ui/issues/3413
* https://github.com/zwave-js/zwave-js-ui/issues/3313

**Comment by myoung34** [comment]
Seems to be related to this ![image](https://github.com/zwave-js/zwave-js-ui/assets/879922/58692caf-dac7-44bf-bdd0-cd300a73d078)

**Comment by robertsLando** [comment]
I'm aware of this guys and I'm sorry for the inconvenience but I don't understand what's going on, it's an issue with zui but seems something related to github actions/docker or yarn....

**Comment by myoung34** [comment]
The builds are timing out if you look at the error. Your best bet is to build them in parallel similar to how [I do](https://github.com/myoung34/docker-github-actions-runner/actions/runs/6898384474)

it's basically an update to your [yaml](https://github.com/myoung34/docker-github-actions-runner/blob/master/.github/workflows/deploy.yml)

**Comment by robertsLando** [comment]
What I sincerly do not understand is why they always worked and then suddenly stopped working. It happened to me on another project with the same result, with armv6/v7 install simply hangs for no reason. Notice that the test workflows (each test runs on a specific arch) end successfully so the error happens only when building all archs together with buildx. This makes me think something may have changed on action runner env

**Comment by myoung34** [comment]
It could be, but the blast radius would be smaller in parallel (they'd upload as they finish instead of failing at the very end for all if one does)

You could always try qemu/buildx instead of `uraimo/run-on-arch-action@v2` and see if that helps

**Comment by robertsLando** [comment]
@myoung34 Wait, it's not `run-on-arch-action` that fails, it's docker build-push action

**Comment by myoung34** [comment]
[This run failed to build and push](https://github.com/zwave-js/zwave-js-ui/actions/runs/6891324377) for a 2 hour timeout and comes from [this workflow file](https://github.com/zwave-js/zwave-js-ui/blob/master/.github/workflows/docker-release.yml#L14) but youre right, i was looking at the wrong file in my previous. but you could still split them up so they all build simultaneously similar to mine

**Comment by robertsLando** [comment]
I already saw that and seems related to yarn, but what I don't get is firstly why it suddenly stopped working, second why that error is not always shown, it appeared only on some logs not all... sometimes build fail others it hangs, I will do more tries on Monday as I'm busy this weekend but if someone finds a solution feel free to open a PR. Sorry for the inconvenience guys

**Comment by robertsLando** [comment]
I have a feel it's something related to yarn install eating all memory on arm32: https://github.com/yarnpkg/berry/issues/3972.

x64 takes few seconds to complete, arm64 takes around 5 minutes and arm32 just hangs...

**Comment by robertsLando** [comment]
What I tried:

- Upgrade yarn to v4 (previously was 3.2.1)
- Pre-install all deps before running buildx, remove node_modules from dockerignore and inside docker image skip the yarn install if node modules folder is present. The hang now happens on yarn rebuild command (needed for serialport)
- Tried to upgrade all gh actions to latest
- Checked if runner version changed between last succeed image and the first errored one, seems to be the same: 2.311.0 

Failed: https://github.com/zwave-js/zwave-js-ui/actions/runs/6891324352/job/18756410049#step:1:1
Success: https://github.com/zwave-js/zwave-js-ui/actions/runs/6888473577/job/18737546618#step:1:1

- Tried changing docker base image, no success. Could be related to alpine maybe and may be worth tring with a non-alpine image but that will require to drop armv6 support.
- Tried to add monitoring to see if the hang is due to missing cpu/ram resources on github runner, seems not a problem: https://github.com/zwave-js/zwave-js-ui/actions/runs/6960408785#summary-18939826176
- Tried to increeese nodejs memory https://github.com/zwave-js/zwave-js-ui/commit/386d098fde87f49e3ad4d94a323a3fed1aa11589#diff-f34da55ca08f1a30591d8b0b3e885bcc678537b2a9a4aadea4f190806b374ddcR22
- To try: try to run buildx locally to see if it hangs there too, if so

**Comment by getlarge** [comment]
@robertsLando I don't know if they are enabled in this organization, but have you tried [larger runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners) ?

**Comment by robertsLando** [comment]
@getlarge already checked and I don't have them, dunno if they could be enabled by @AlCalzone on organization settings?

**Comment by chrisns** [comment]
thats a pay for feature unless github are being super generous

**Comment by chrisns** [comment]
parallel building with a runner for each arch would be lovely but unless someone has an easy suggestion on how to fan out and fan back in to push a single (multi arch) docker image it doesn't really work, though would provide a full runner's compute resource to each arch build

**Comment by robertsLando** [comment]
I think a way could be to export the image as asset and import all them in the final step and push all them together

**Comment by robertsLando** [comment]
Ok nice finding... using `npm rebuild` in place of `yarn rebuild` has reduced build time for arm64 from 360s to 150s. Let's see how it goes for arm32..

**Comment by snowmangh** [comment]
Hello. Looks like this issue has been resolved? I just tried and getting the same error 400.

**Comment by chrisns** [comment]
400 on trying to pull from docker hub?
What Arch?
I'd generally suggest using ghcr instead of docker to avoid rate caps etc

**Comment by snowmangh** [comment]
@chrisns Yes, from **docker.io** zwavejs/zwave-js-ui:latest. I always used this method without issues, so far via Portainer.

**What Arch?** Apollolake

**Comment by snowmangh** [comment]
Interesting. Just noticed that I am pointing to this 2-week old image which may explain my issue.

![image](https://github.com/zwave-js/zwave-js-ui/assets/90811206/5b5d61df-b84d-4427-8473-0ad5a81e994b)
